homeassistant/components/nuki/binary_sensor.py

```diff
@@ -50,14 +50,6 @@ def unique_id(self) -> str:
         """Return a unique ID."""
         return f"{self._nuki_device.nuki_id}_doorsensor"
 
-    @property
-    def extra_state_attributes(self):
-        """Return the device specific state attributes."""
-        data = {
-            ATTR_NUKI_ID: self._nuki_device.nuki_id,
```

Comment:
If we remove the attribute it's a breaking change. If we want to deprecate it first, we need to keep the attribute here.

But we don't currently have a good way of notifying users about state attributes that are deprecated.

Reply:
Is it possible to have a note in the release notes to warn users? So at least if they read the notes (which probably only a small fraction of users does) they have a chance to change their automations ..

Reply:
Do you mean to add a note in the release notes about the deprecated attributes, and don't remove them yet?

Yes, that's possible. It's probably best that you check the release notes after the beta cut for the April release and make a PR to the rc branch of the user docs at that time, since there won't be a merged core PR for us to track.

Reply:
What we could do is to just add comments above the deprecated attributes in this PR, saying when they are to be removed. Then we'd have a PR to track for the release notes.

Reply:
Okay so I add them again and add a comment in the code?
What about the Nuki ID? Should we remove it now? Guess nobody is using it in an automation 🤷🏻‍♂️ Then remove the battery critical attributes in 2024.9 or just remove all of them later and nothing now?

Reply:
Yes I like the comment approach. Please also leave the nuki id for now to have it consistent.

Reply:
I've added the comments now.